types/v56/vm_types.go

```diff
@@ -129,14 +142,15 @@ type ComputePolicy struct {
 
 // CreateVmParams is used to create a standalone VM without a template
 type CreateVmParams struct {
-	XMLName     xml.Name   `xml:"CreateVmParams"`
-	XmlnsOvf    string     `xml:"xmlns:ovf,attr"`
-	Xmlns       string     `xml:"xmlns,attr,omitempty"`
-	Name        string     `xml:"name,attr,omitempty"`    // Typically used to name or identify the subject of the request. For example, the name of the object being created or modified.
-	PowerOn     bool       `xml:"powerOn,attr,omitempty"` // True if the VM should be powered-on after creation. Defaults to false.
-	Description string     `xml:"Description,omitempty"`  // Optional description
-	CreateVm    *Vm        `xml:"CreateVm"`               // Read-only information about the VM to create. This information appears in the Task returned by a createVm request.
-	Media       *Reference `xml:"Media,omitempty"`        // Reference to the media object to insert in the new VM.
+	XMLName     xml.Name     `xml:"CreateVmParams"`
+	XmlnsOvf    string       `xml:"xmlns:ovf,attr"`
+	Xmlns       string       `xml:"xmlns,attr,omitempty"`
+	Name        string       `xml:"name,attr,omitempty"`    // Typically used to name or identify the subject of the request. For example, the name of the object being created or modified.
+	PowerOn     bool         `xml:"powerOn,attr,omitempty"` // True if the VM should be powered-on after creation. Defaults to false.
+	Description string       `xml:"Description,omitempty"`  // Optional description
+	CreateVm    *Vm          `xml:"CreateVm"`               // Read-only information about the VM to create. This information appears in the Task returned by a createVm request.
+	Media       *Reference   `xml:"Media,omitempty"`        // Reference to the media object to insert in the new VM.
+	BootOptions *BootOptions `xml:"BootOptions,omitempty"`  // Used to specify VM's Boot Options on creation
```

Comment:
Could this field also be used in `InstantiateVmTemplateParams` (used for Standalone VM creation from template) and `ReComposeVAppParams` (used in vApp VM creation from template)? 

Or is it not relevant for templated VMs?

Reply:
It is not relevant from what I checked, according to `RecomposeVappParams` type in the API [1], it has the field `ReconfigureItem` to reconfigure a VM but from my understanding we don't have that implemented on purpose.

Also in the UI, you can only choose the Boot options/Firmware if you create an empty VM instead of a template.

[1] - https://developer.vmware.com/apis/1703/vmware-cloud-director/doc/doc/types/RecomposeVAppParamsType.html